warp/transpiler/Operations/Bitwise.py

```diff
@@ -32,44 +32,47 @@ def __init__(self):
         super().__init__(operator.xor, UINT256_MODULE, "uint256_xor")
 
 
-class Shl(Binary):
-    def evaluate_eagerly(self, b, x):
-        return (x << b) % 2**256
+class Shl(Binary):  # ARG ORDER NOT CONVENTIONAL
+    def evaluate_eagerly(self, shift, value):
+        return value << shift
 
     def generate_cairo_code(self, op1, op2, res):
-        # the reverse order in `uint256_shr` call is intentional
         return [f"let (local {res} : Uint256) = uint256_shl({op2}, {op1})"]
 
+    @classmethod
     def required_imports(cls):
         return {UINT256_MODULE: {"uint256_shl"}}
 
-class Shr(Binary):
-    def evaluate_eagerly(self, b, x):
-        return x >> b
+
+class Shr(Binary):  # ARG ORDER NOT CONVENTIONAL
+    def evaluate_eagerly(self, shift, value):
+        return value >> shift
 
     def generate_cairo_code(self, op1, op2, res):
-        # the reverse order in `uint256_shr` call is intentional
         return [f"let (local {res} : Uint256) = uint256_shr({op2}, {op1})"]
 
+    @classmethod
     def required_imports(cls):
         return {UINT256_MODULE: {"uint256_shr"}}
 
 
-def sar(a, b):
-    b = uint256_to_int256(b)  # per yellow paper, only b is treated as signed
-    return int256_to_uint256(a >> b)
+def sar(value, shift):
```

Comment:
Why isn't sar function in Sar class?

Reply:
It doesn't need the class state and semantically it is just an operation on two numbers